### DISCUSSION TITLE
Slash sub commands and sub command groups

### DISCUSSION THREAD

**Discussion by skylarr1227** [InterStella0/starlight-dpy#6] [Q&A] [answered]
So, it seems only the top command in a group command (the useless parent that does nothing) shows up for app_commands & hybrid groups both. 

Is there a way to show these, maybe I'm missing where you mention it but i don't see it mentioned.

**Comment by InterStella0** [comment, marked as answer]
It is intended to only show the parent commands since I find it redundant to show their subcommand given that you can see it on the help command group. Maybe I should add a kwargs to change it

**TL;DR replies resulted in final answer**
Do your own formatting since the library only shows parent commands.
```py
class MyMenuHelpCommand(starlight.MenuHelpCommand):
    async def format_cog_page(view: starlight.HelpMenuCog, cmds: List[commands.Command]):
        embed = discord.Embed()
        # do my logic here
        return embed
```
This format is used in both MenuHelpCommand.send_bot_help and send_cog_help
References: [HelpMenuCog](https://starlight-dpy.readthedocs.io/en/latest/help-command.html#starlight.HelpMenuCog), [Command](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.Command), [format_cog_page](https://starlight-dpy.readthedocs.io/en/latest/help-command.html#starlight.MenuHelpCommand.format_cog_page)

**Reply by skylarr1227** [reply]
Im not sure what you mean, heres what mine shows
![image](https://user-images.githubusercontent.com/44381751/211449893-0e1f6c82-0339-4a66-83f5-928cfa38ca91.png)
![image](https://user-images.githubusercontent.com/44381751/211449912-70617407-61c9-4fd1-a1aa-7adbdc6c1820.png)
![image](https://user-images.githubusercontent.com/44381751/211449928-80a3a1da-aee0-4970-8a6f-64c991c5f70b.png)

In this example, yes it would be i suppose....

But in others no not at all....such as this
![image](https://user-images.githubusercontent.com/44381751/211450312-b26c716e-89bc-43d2-b24f-92c2f166ce2e.png)
![image](https://user-images.githubusercontent.com/44381751/211450512-520cb5ee-e29b-43d9-be9e-80a5625e9dd9.png)

Most of our bots commands are being moved into groups like this, since we were at 100 commands, the max... Im trying to knock that down to 50 or so, using subs. So it would be helpful for help to show :P

**Reply by InterStella0** [reply]
Well, truth is the commands are shown based on `HelpCommand.get_bot_mapping` after it has been filtered so you can blame d.py for that, if you want to show more than just the parent, or change the way it is shown in general, you can simply modify it in `MenuHelpCommand.format_cog_page` by overriding it when you subclass [MenuHelpCommand](https://starlight-dpy.readthedocs.io/en/latest/help-command.html#menuhelpcommand).

```py
class MyMenuHelpCommand(starlight.MenuHelpCommand):
    async def format_cog_page(view, cmds):
        embed = discord.Embed()
        # do my logic here
        return embed
```
This is because every cog in send_bot_help is using format_cog_page to reduce redundancy. At the same time, send_cog_help is also using format_cog_page so you can have an equal view.
All you have to do is use `Group.walk_commands` everytime you see a Group object. The returning embed will be shown on the message.

There is no direct way to do this as of now, and I don't think I will add the keyword argument since user can easily change it

**Reply by skylarr1227** [reply]
Thats totally understandable! Is there any way i could convince you to contact me somewhere I can more interactively ask you some questions about this as i try to make that work? Im not the best dev in the world, and it would just be very helpful! 

if discord is an option, again mine is skylarr#6666-I wouldnt bug you to much promise.

**Reply by InterStella0** [reply]
No, but you can ask in the discord.py server for general dpy help since you have a list of `Command` objects in `format_cog_page` and all you have to do is do some formatting. This is still using discord.py and not a fork, it's just a utility library. I'm trying to keep every question goes in GitHub so other people can see the answers instead of individually helping with basic codes. 
You can read the [docs](https://starlight-dpy.readthedocs.io/en/latest/help-command.html) and [dpy docs](https://discordpy.readthedocs.io/en/stable/).
